README.md

```diff
@@ -204,66 +206,112 @@ const toast = document.createElement('std-toast');
+  that is not [interactive content](https://html.spec.whatwg.org/#interactive-content-2).
+  These children provide the toast's message.
+- There may be zero or one actions,
+  which are either `<a>` or `<button>` elements,
+  decorated with the `slot="action"` attribute.
 
-TODO([#17](https://github.com/jackbsteinberg/std-toast/issues/17)): what about title or icon? They should potentially also be accomodated, in a similar fashion.
+TODO([#17](https://github.com/jackbsteinberg/std-toast/issues/17)): what about title or icon?
+They could potentially also be accomodated, in a similar fashion.
 
-Thus, the following would all work well out of the box:
+Thus,
+the following would all work well out of the box,
+and be comformant to the content model:
 
 ```html
 <std-toast>Hello world!</std-toast>
 
-<std-toast><p>Hello world!</p></std-toast>
+<std-toast><p>Hello <strong>world</strong>!</p></std-toast>
 
-<std-toast>
-  <p>Hello world!</p>
-  <button>Click me!</button>
+<std-toast id="toast3">
+  Hello world!
+  <button slot="action" class="fancy-button">Click me!</button>
 </std-toast>
 
 <std-toast>
-  <p>Hello world!</p>
-  <a href="https://example.com/" target="_blank">Click me!</button>
+  Hello world!
+  <a slot="action" href="https://example.com/" target="_blank">Click me!</button>
 </std-toast>
-```
 
-TODO([#21](https://github.com/jackbsteinberg/std-toast/issues/21)): How will this pattern support work well?
+<script type="module">
+document.querySelector("#toast3").action.onclick = e => { /*...*/ };
+</script>
+```
 
-These can equivalently be created via JavaScript: ([#12](https://github.com/jackbsteinberg/std-toast/issues/12))
+These can equivalently be created via JavaScript:
 
 ```js
-// TODO: write this once we figure out the JS API for supplying HTML messages,
-// and for supplying customizable actions
+const toast1 = showToast("Hello world!");
+
+const toast2 = showToast();
+toast.innerHTML = `<p>Hello <strong>world</strong>!</p>`;
+
+const toast3 = showToast("Hello world!", { action: "Click me!" });
+toast3.action.className = "fancy-button";
+toast3.action.onclick = e => { /*...*/ };
+
+const toast4 = showToast("Hello world!", { action: document.createElement("a") });
+Object.assign(toast4.action,
+  href: "https://example.com",
+  target: "_blank",
+  textContent: "Click me!"
+});
+
+const toast4SetterDemo = showToast("Hello world!");
+toast4SetterDemo.action = document.createElement("a");
+Object.assign(toast4SetterDemo.action, /* as before */);
 ```
 
+(Note: because frames are only painted after JavaScript runs to completion,
+the setup code in the lines immediately after `showToast()` will be applied before the toast is ever shown to the user.)
+
 More complex toasts,
 that don't fit the above content model,
-would require more custom handling on the part of the developer:
+would require custom handling on the part of the developer,
+as they are stepping outside of the supported use case:
```

Comment:
Do we have accessibility backup for why those are the supported ones, or is it just that we want to reduce complexity?

Reply:
I'm not sure if this is an accessibility thing, as much as a "what is a toast" thing.